src/Workspaces/Core/Portable/CodeGeneration/Symbols/CodeGenerationTypeSymbol.cs

```diff
@@ -38,7 +38,7 @@ public virtual ImmutableArray<INamedTypeSymbol> Interfaces
         public ImmutableArray<INamedTypeSymbol> AllInterfaces
             => ImmutableArray.Create<INamedTypeSymbol>();
 
-        public bool IsReferenceType => false;
+        public bool IsReferenceType => TypeKind != TypeKind.Enum && TypeKind != TypeKind.Struct && TypeKind != TypeKind.Error;
```

Comment:
Unless there is a reason for the need of this change, I'd keep it as is, and use TypeKind.Class in the places `IsReferenceType` is used in this PR. It's always used as a second condition for "IsRecord", so we know for sure it's either a class or a struct.

I also believe IsRecord is checked along with TypeKind in the compiler layer, so that will be consistent as well.

Reply:
I can live with that ;-)

---
In reply to: [609297016](https://github.com/dotnet/roslyn/pull/52464#discussion_r609297016) [](ancestors = 609297016)